app/lib/hooks/useA11yErrorAnnouncement.tsx

```diff
@@ -1,28 +1,38 @@
 import { useRef } from 'react';
 import { AccessibilityInfo } from 'react-native';
+import { FieldErrorsImpl } from 'react-hook-form';
 
 import { useDebounce } from '../methods/helpers';
 import { textInputDebounceTime } from '../constants';
 
 interface IUseA11yErrorAnnouncement {
-	error: string | undefined;
+	errors: FieldErrorsImpl<any>;
 }
 
-const useA11yErrorAnnouncement = ({ error }: IUseA11yErrorAnnouncement) => {
-	const previousMessage = useRef<string>(error);
-	const announced = useRef<boolean>(false);
-	const shouldAnnounce = error && error !== previousMessage.current && !announced.current;
-
+const useA11yErrorAnnouncement = ({ errors }: IUseA11yErrorAnnouncement) => {
+	const previousMessages = useRef<FieldErrorsImpl<any>>(errors);
+	const announced = useRef<Record<string, boolean>>({});
 	const handleA11yAnnouncement = useDebounce(() => {
-		if (shouldAnnounce) {
-			const message = (error || '').trim();
-			if (message) {
+		const hasError = Object.keys(errors).length;
+		if (!hasError) {
+			announced.current = {};
+			previousMessages.current = {};
+			return;
+		}
+
+		Object.entries(errors).forEach(([fieldName, error]: [string, any]) => {
+			const message = error?.message?.trim();
+
+			if (message && message !== previousMessages.current[fieldName] && !announced.current[fieldName]) {
```

Comment:
I'm not sure it works like that, but here it goes.

- User enters an invalid email
- App announces the error
- User enters another invalid email
- Is the app going to announce it again? It should, right?